packages/astro/src/core/build/generate.ts

```diff
@@ -257,24 +257,39 @@ async function generatePage(
 		mod: pageModule,
 	};
 
+	// Get paths for the route, calling getStaticPaths if needed.
+	const paths = await getPathsForRoute(pageData, pageModule, opts, builtPaths);
+
+	// We do this here to get the stat `filteredPaths / paths` for info logging.
+	const filteredPaths = paths.filter((path) => opts.settings.config.build.filterPage(path));
+
+	// Cleanup entries added from `getPathsForRoute`
+	const skippedPaths = paths.filter((path) => !filteredPaths.includes(path));
+	skippedPaths.forEach((path) => builtPaths.delete(removeTrailingForwardSlash(path)));
+
+	const routesSkipped =
+		filteredPaths.length === 0
+			? dim('[skipped fully]')
+			: filteredPaths.length < paths.length
+			? dim(`[skipped ${paths.length - filteredPaths.length} paths of ${paths.length}]`)
+			: '';
+
```

Comment:
the `.filter` call could be moved later (at L339) if there's no value in displaying the `(filtered/total)` stat in the logs, which would make the chunk of L266-L268 and L273-L274 useless.

Reply:
with a `pagePath => pagePath.startsWith('/fr')`, the render looks like this:

```
▶ src/pages/blog/category/[...path].astro [skipped fully]
▶ src/pages/blog/[...path].astro [skipped fully]
▶ src/pages/[...path].astro [skipped 161 paths of 191]
  ├─ /fr/index.html (+2.28s)
^C
```